pywhy_stats/kci.py

```diff
@@ -0,0 +1,196 @@
+from typing import Callable
+
+import numpy as np
+from numpy import shape, ndarray
+from numpy.linalg import eigvalsh
+from numpy.random import permutation
+from scipy import stats
+
+from pywhy_stats.custom_kernels import rbf_kernel
+from pywhy_stats.pvalue_result import PValueResult
+
+
+def ind(X: np.ndarray,
+        Y: np.ndarray,
+        kernel_X: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
+        kernel_Y: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
+        approx: bool = True,
+        null_ss: int = 1000) -> PValueResult:
+    """
+    Python implementation of Kernel-based Conditional Independence (KCI) test and the unconditional version.
+    The original Matlab implementation can be found in http://people.tuebingen.mpg.de/kzhang/KCI-test.zip
+
+    References
+    ----------
+    [1] K. Zhang, J. Peters, D. Janzing, and B. Schölkopf,
+    "A kernel-based conditional independence test and application in causal discovery," In UAI 2011.
+    [2] A. Gretton, K. Fukumizu, C.-H. Teo, L. Song, B. Schölkopf, and A. Smola, "A kernel
+    Statistical test of independence." In NIPS 21, 2007.
```

Comment:
You can use the Sphinx bibtex extension, which replaces these with `.. footbibliography::` and then you can reference them in any function as `:footcite:`<name>``. 

See: https://github.com/py-why/pywhy-stats/blob/main/doc/references.bib
and FisherZTest, which shows the pattern.

This just makes the citations of stuff more consolidated especially for papers that are referenced across multiple functions/classes.